### PR TITLE
Validate entity property names

### PR DESCRIPTION
**Pull request by javiereguiluz**
This refactores #394.

### CODE REVIEW

File: Command/Validators.php [hunk 1]
```diff
@@ -127,6 +127,19 @@ public static function validateEntityName($entity)
         return $entity;
     }
 
+    public static function validateEntityPropertyName($property)
+    {
+        if (in_array($property, self::getReservedWords())) {
+            throw new \InvalidArgumentException(sprintf('Name "%s" is a PHP reserved word.', $property));
+        }
+
+        if (!preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $property)) {
```
Comment (OskarStark) [inline Command/Validators.php]: if you use the pattern directly, you can remove the method from the generator class

File: Command/Validators.php [hunk 2]
```diff
@@ -127,6 +127,19 @@ public static function validateEntityName($entity)
         return $entity;
     }
 
+    public static function validateEntityPropertyName($property)
+    {
+        if (in_array($property, self::getReservedWords())) {
```
Comment (OskarStark) [inline Command/Validators.php]: if we use this, we shall remove the isReservedKeyword()-method, but this is a BC

**Comment by OskarStark** [comment]
looks like the is_null() check is necessary, because travis is running and running :)

**Comment by OskarStark** [comment]
my PR #394 passed all the test now :)

**Comment by OskarStark** [comment]
@javiereguiluz that is the same problem i had on my local machine

bildschirmfoto 2015-07-22 um 10 36 58

i don't know exactly why, but if you change this:
```php
if (!preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $property)) {
```

to this:
```php
if (!is_null($property) && !preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $property)) {
```

it should be working!

**Comment by javiereguiluz** [comment]
Closing it in favor of #394.

**Pull request closed without merging**